### ISSUE TITLE
ConcurrentModificationException in AbstractStyledTextOutput

### ISSUE THREAD

**Issue by DanielThomas** [gradle/gradle#7508]
```
14:02:27 Caused by: java.util.ConcurrentModificationException
14:02:27    at org.gradle.internal.logging.text.AbstractStyledTextOutput.format(AbstractStyledTextOutput.java:47)
14:02:27    at org.gradle.internal.logging.text.AbstractStyledTextOutput.formatln(AbstractStyledTextOutput.java:58)
14:02:27    at org.gradle.api.tasks.diagnostics.internal.PropertyReportRenderer.addProperty(PropertyReportRenderer.java:30)
14:02:27    at org.gradle.api.tasks.diagnostics.PropertyReportTask.generate(PropertyReportTask.java:48)
14:02:27    at org.gradle.api.tasks.diagnostics.AbstractReportTask$2.generateReport(AbstractReportTask.java:75)
14:02:27    at org.gradle.api.tasks.diagnostics.internal.ReportGenerator.generateReport(ReportGenerator.java:58)
14:02:27    at org.gradle.api.tasks.diagnostics.AbstractReportTask.generate(AbstractReportTask.java:85)
14:02:27    at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
14:02:27    at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
14:02:27    at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
14:02:27    at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
14:02:27    at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:786)
14:02:27    at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:753)
14:02:27    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
14:02:27    at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
14:02:27    at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
14:02:27    at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
14:02:27    at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
14:02:27    at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
14:02:27    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
14:02:27    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
14:02:27    ... 31 more
```

### Steps to Reproduce (for bugs)
Unknown

### Your Environment
Gradle 4.9

**Comment by big-guy** [comment]
@DanielThomas do you know which tasks were being run when this happened? Is this a multi-project build?

**Comment by DanielThomas** [comment]
At the very least the property report task. Yes, it's a multi-project build, with `org.gradle.parallel=true`.

**Comment by rpalcolea** [comment]
This still unresolved :) let's open it again

**Comment by alpar-t** [comment]
The issue was added to the [7.3.0 release notes](https://docs.gradle.org/7.3/release-notes.html) but it has no linked PR, was this actually fixed ?

**Comment by mlopatkin** [comment]
It was fixed in #17658.
